test/scripts/generate_big_message.sh

```diff
@@ -109,8 +109,24 @@ esac
 generate_message >"$FILE"
 
 #asserting the message has the correct size
-if test "$SIZE" != "$(stat -c%s "$FILE")"
-then
-    rm -f "$FILE"
-    exit 1
-fi
+# FreeBSD stat uses "-f" for output formatting, GNU uses "-c"
+# FreeBSD stat use "%z" for total size in bytes, GNU uses "%s"
+case $(uname -s) in
+  Linux )
+    if test "$SIZE" != "$(stat -c%s "$FILE")"
+    then
+        rm -f "$FILE"
+        exit 1
+    fi
+  ;;
+  FreeBSD )
+    if test "$SIZE" != "$(stat -f%z "$FILE")"
+        then
+            rm -f "$FILE"
+            exit 1
+    fi
+  ;;
+  *)
```

Comment:
This cannot just handle Linux and FreeBSD and bail on everything else.

I suggest testing stat itself, not uname, to determine what stat is available. Something like

    if stat --version | grep 'GNU coreutils' >/dev/null; then
      # use the GNU syntax
    elif
      ...
    fi